src/core/server/http/logging/get_response_log.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import querystring from 'querystring';
+import { isBoom } from '@hapi/boom';
+import type { Request } from '@hapi/hapi';
+import numeral from '@elastic/numeral';
+import { LogMeta } from '@kbn/logging';
+import { EcsEvent, Logger } from '../../logging';
+import { getResponsePayloadBytes } from './get_payload_size';
+
+const ECS_VERSION = '1.7.0';
+const FORBIDDEN_HEADERS = ['authorization', 'cookie', 'set-cookie'];
+const REDACTED_HEADER_TEXT = '[REDACTED]';
+
+// We are excluding sensitive headers by default, until we have a log filtering mechanism.
+function redactSensitiveHeaders(
+  headers: Record<string, string | string[]>
+): Record<string, string | string[]> {
+  return (
+    headers &&
+    Object.keys(headers).reduce(
+      (acc, key) => ({
+        // Create a shallow copy to prevent mutating the original headers
+        ...acc,
```

Comment:
super nit: The creation of a new object on every iteration creates high GC pressure. What if we do it in this way?
```js
const FORBIDDEN_HEADERS = new Set(['authorization', 'cookie', 'set-cookie']);
function redactSensitiveHeaders(
  headers?: Record<string, string | string[]>
): Record<string, string | string[]> {
  const result = {} as Record<string, string | string[]>;
  if (headers) {
    for (const key of Object.keys(headers)) {
      result[key] = FORBIDDEN_HEADERS.has(key) ? REDACTED_HEADER_TEXT : headers[key];
    }
  }
  return result;
}

```

Reply:
Agreed, this is much better. Will update